templates/deployments.yaml

```
@@ -1377,6 +1377,13 @@ spec:
         image: '{{ .Values.st2chatops.image.repository | default "stackstorm" }}/{{ .Values.st2chatops.image.name | default "st2chatops" }}:{{ tpl (.Values.st2chatops.image.tag | default .Chart.AppVersion) . }}'
         imagePullPolicy: {{ .Values.st2chatops.image.pullPolicy | default .Values.image.pullPolicy }}
         env:
+        {{- if index .Values.st2chatops "api_key" }}
+        - name: ST2_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Release.Name }}-st2-auth
+              key: chatops_api_key
+        {{- else }}
```

I suppose we could do this instead:
```suggestion
        {{- if index .Values.st2chatops.env "ST2_API_KEY" }}
```
Would that be better?